src/Autonomous.cpp

```diff
@@ -103,6 +131,18 @@ void Autonomous::setNavState(NavState s)
   {
     log(LOG_INFO, "Changing navigation state to %s\n", NAV_STATE_NAMES[nav_state]);
     plan_cost = INFINITE_COST;
+	if (currentLegDone())
+	{
+		// Remove previous leg
+		urc_targets.pop();
+
+		// clear the cached data points
+		leftPostFilter.reset();
+		rightPostFilter.reset();
```

Comment:
As discussed below, let's remove the `currentLegDone()` method and move these three statements to the `startNextLeg()` method. Then I think we can get rid of this block of code completely, which simplifies things.